sockeye/inference.py

```diff
@@ -1043,25 +1047,58 @@ def _beam_search(self,
             if self.restrict_lexicon:
                 best_word_indices[:] = vocab_slice_ids.take(best_word_indices)
 
-            # (4) get hypotheses and their properties for beam_size winning hypotheses (ascending)
+            # (4) Normalize the scores of newly finished hypotheses
+            newly_finished = (best_word_indices == self.vocab_target[C.EOS_SYMBOL]).expand_dims(axis=1)
```

Comment:
shouldn't an hypothesis also be finished if it produces a pad symbol? at least this is the existing logic below.

Reply:
These are ones that have just been completed on this timestep.

Reply:
can you elaborate? Not sure I understand your reply.

Reply:
This gets only the items for which EOS was generated on the current timestep (not ones that were already finished on previous time steps), so that normalization can be applied. The pad symbol can only be produced after EOS is generated, right?

Reply:
no, why would that be the case? A model can generate any token in the target vocabulary, including PAD.